lib/pure/pegs.nim

```diff
@@ -2040,6 +2058,8 @@ proc escapePeg*(s: string): string =
 when isMainModule:
   assert escapePeg("abc''def'") == r"'abc'\x27\x27'def'\x27"
   assert match("(a b c)", peg"'(' @ ')'")
+  static:
+    assert match("(a b c)", peg"'(' @ ')'")
```

Comment:
Here is an example how to lift a test to be executed at compile time as well as runtime:
https://github.com/nim-lang/Nim/blob/devel/tests/stdlib/tjsonmacro.nim#L638